crates/language/src/language_settings.rs

```diff
@@ -109,12 +109,23 @@ pub struct LanguageSettings {
     pub code_actions_on_format: HashMap<String, bool>,
 }
 
-/// The settings for [GitHub Copilot](https://github.com/features/copilot).
+/// The provider that supplies inline completions.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum InlineCompletionProvider {
+    None,
+    #[default]
+    Copilot,
+    Supermaven,
+}
+
+/// The settings for inline completions, such as [GitHub Copilot](https://github.com/features/copilot)
+/// or [Supermaven](https://supermaven.com).
 #[derive(Clone, Debug, Default)]
-pub struct CopilotSettings {
-    /// Whether Copilot is enabled.
-    pub feature_enabled: bool,
-    /// A list of globs representing files that Copilot should be disabled for.
```

Comment:
We need to include `feature_enabled` in the `InlineCompletionSettings` since `copilot` settings now aliases to this struct.